agents/TabularRTabularQ.py

```diff
@@ -0,0 +1,17 @@
+import numpy as np
+from agents.TabularQ import TabularQ
+from bayesianapproximator import TabularBayesianApproximation
+
+class TabularRTabularQ(TabularQ):
+    def __init__(self, state_shape, num_acts):
+        super().__init__(state_shape, num_acts)
+
+        self.rewardApprox = TabularBayesianApproximation(state_shape, num_acts)
+        self.epsilon = 0.05
+
+    def update(self, s, sp, r, a, done):
+        self.rewardApprox.update_stats(s, a, r)
+        samples = self.rewardApprox.sample(s, a, 1)
+
+        bonus = samples[0]
```

Comment:
The `max(samples) - mean(samples)` thing doesn't actually work here, since `max(samples) == mean(samples)` this will mean the bonus is always 0. We could probably do some refactoring to fix this, but for now it wasn't worth it to me.